src/estudiante.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass, field
+from calendario import Calendario
+
+@dataclass
+class Estudiante:
+    id: int
+    nombre: str
+    calendario: Calendario
+    id_de_horarios: list = field(default_factory=list) # lista de ids de horarios
+
+    def inscribir_horario(self, horario):
```

Comment:
Look for more information to make sure handling all scenarios should be in this method or should raise an error if can't add the schedule

Reply:
Looks like we can use a behavioral pattern called Observer to track inscriptions and update all objects
Here some reference:
- https://youtu.be/tAuRQs_d9F8?t=218
- https://refactoring.guru/design-patterns/observer#:~:text=Observer%20is%20a%20behavioral%20design,the%20object%20they're%20observing.

Reply:
Looks like we don't need this because the pub/sub pattern shines when we need a notification for specific users. In this case, we are not doing such a thing, just assigning a relationship but not sending an event like a notification from one to another.